2.18.1 viewer

mstucky opened the pull request:
Contents of branch to be merged to Master:
#173822671 Add validation to HeatmapDataGenerator to prevent 2 covars on the same axis with the same name.
#173967727 embedded expandable maps do not allow downloads.
#173967889 Covariate search category display sort bug
#173967925 REFLECT website not displaying properly after implementing 2.18.0
#173968107 Modify matrix selections feature to display and function when items on only one axis are selected.
#174023319 Dendro selection box not in correct location in ribbon mode if map for large maps
#174023365 Search box no longer finding labels containing the period (.) character

marohrdanz commented:
I looked over the code just to see which changes corresponded to which PT stories mentioned in the pull request. I listed what I think is the case below--please let me know if these are incorrect. (This exercise was as much for my understanding as anything else.)

One question @mstucky : which PT story are the changes in PdfGenerator.js for?

```
#173822671 Add validation to HeatmapDataGenerator to prevent 2 covars on the same axis with the same name.
```
I think this is the setClassName function added in InputClass.java.

```
#173967727 embedded expandable maps do not allow downloads.
```
I think this is the change in NGCHM\_Util.js: adding 'allow-downloads' on line 1199. 

```
#173967889 Covariate search category display sort bug
```
I think this is the changes to NgChm.UTIL.createCheckBoxDropDown.

```
#173967925 REFLECT website not displaying properly after implementing 2.18.0
```
I think this is the changes to NGCHM.css to make the classes/ids more ngchm-specific, and the corresponding changes in the javascript that use these classnames and ids. 

```
#173968107 Modify matrix selections feature to display and function when items on only one axis are selected.
```
I think this change is most of the changes in DetailHeatMapDisplay.js (that are not part of the fix for #174023319), Linkouts.js, and MatrixManager.js.

```
#174023319 Dendro selection box not in correct location in ribbon mode if map for large maps
```
Is this just the line 652 in DetailheatMapDisplay.js? If so, wow, simple fix! 

```
#174023365 Search box no longer finding labels containing the period (.) character
```
I think this is the change in SearchManager.js on lines 172 & 173.

mstucky commented:
Hi Mary,

You are very perceptive and hit the nail on the head on all the changes.

The PdfGenerator changes are for:  _#173968107 Modify matrix selections feature to display and function when items on only one axis are selected._ This is because I wrote an asynchronous process for downloading needed tiles in order to wait for them when trying to do a selection of all rows/cols on an axis.  Previously for 2.18.0, we modified the PdfGenerator process to add a 3 second wait for all tiles to load in order to build the Data Matrix Distribution.  This involved setting a read window to pull all tiles.  I replaced that timer based solution with the async one created for matrix selections.

marohrdanz commented:
Awesome. Thanks for the info!
